### PR TITLE
[pkg/ottl] Add ability to parse conditions

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:** 
This PR adds a new public API to support parsing and using conditions. 

This implementation opted to add ParseConditions directly to the existing Parser instead of creating a second Parser struct.  Since any Parser needs the Context, Functions, PathExpressionParser, EnumParser, and telemetry settings the 2 structs would be incredibly similar.

See these structs implemented in components here: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/29294

**Link to tracking Issue:** <Issue number if applicable>
Works towards https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/13545

**Testing:** <Describe what testing was performed and which tests were added.>
Added tests

**Documentation:** <Describe the documentation added.>
Added godoc strings

### CODE REVIEW

File: pkg/ottl/parser.go [hunk 1]
```diff
@@ -141,7 +154,50 @@ func (p *Parser[K]) ParseStatement(statement string) (*Statement[K], error) {
...
+// If parsing fails, returns an empty slice  with a multierr error containing
+// an error per failed condition.
+func (p *Parser[K]) ParseConditions(conditions []string) ([]*Condition[K], error) {
+	parsedConditions := make([]*Condition[K], 0, len(conditions))
+	var parseErr error
+
+	for _, condition := range conditions {
+		ps, err := p.ParseCondition(condition)
+		if err != nil {
+			parseErr = multierr.Append(parseErr, fmt.Errorf("unable to parse OTTL condition %q: %w", condition, err))
```
Comment (atoulme) [inline pkg/ottl/parser.go:168]: I think we're trying to move away from multierr - maybe ok to add since it's used already

**Review by atoulme — APPROVED** [review]

File: pkg/ottl/parser.go (hunk 1 above)
Comment (TylerHelmuth) [reply pkg/ottl/parser.go]: Updated

File: pkg/ottl/parser_test.go [hunk 2]
```diff
@@ -1724,6 +1902,65 @@ func Test_parseStatement(t *testing.T) {
...
+			if (err != nil) != tt.wantErr {
+				t.Errorf("parseCondition(%s) error = %v, wantErr %v", tt.condition, err, tt.wantErr)
+				t.Errorf("AST: %+v", ast)
+				return
+			}
+		})
+	}
+}
+
 func Test_Execute(t *testing.T) {
```
Comment (evan-bradley) [inline pkg/ottl/parser_test.go:1964]: Not a big deal, but it probably wouldn't be bad to have a case for `Condition.Eval()`.

**Review by evan-bradley — APPROVED** [review]
Nice!

File: pkg/ottl/parser_test.go (hunk 2 above)
Comment (TylerHelmuth) [reply pkg/ottl/parser_test.go]: Good idea

**Pull request merged**